feed 도메인 리팩토링

seonpilKim opened the pull request:
<!--
✅ Resolve: #이슈번호 형태로 입력해 주세요.
ex) Resolve: #123
-->
## 📌Linked Issues
- Resolve: #161 

<!--
✅ 변경 사항을 자세히 알려주세요. (why -> what -> how)
-->
## ✏Change Details
### Swagger 관련
#### Swagger 설정 추가
```java
@Configuration
public class SwaggerConfig {
	@Bean
	public Docket api() {
		...

		final List<ResponseMessage> responseMessages = new ArrayList<>();
		responseMessages.add(new ResponseMessageBuilder().code(405)
			.message("G002 - 허용되지 않은 HTTP method입니다.").build());
		responseMessages.add(new ResponseMessageBuilder().code(500)
			.message("G001 - 내부 서버 오류입니다.").build());

		return new Docket(DocumentationType.SWAGGER_2)
			...
			.globalResponseMessage(RequestMethod.POST, responseMessages)
			.globalResponseMessage(RequestMethod.GET, responseMessages)
			.globalResponseMessage(RequestMethod.DELETE, responseMessages)
			.globalResponseMessage(RequestMethod.PUT, responseMessages)
			...
	}
...
}
```
- 모든 API에서 발생하는 공통 응답 추가
#### 컨트롤러 메소드마다 `@ApiResponse` 추가
```java
public class PostController {
...
	...
	@ApiResponses({
		@ApiResponse(code = 200, message = "F002 - 게시물 삭제에 성공하였습니다."),
		@ApiResponse(code = 400, message = "G003 - 유효하지 않은 입력입니다.\n"
			+ "F001 - 존재하지 않는 게시물입니다.\n"
			+ "F002 - 게시물 게시자만 삭제할 수 있습니다."),
		@ApiResponse(code = 401, message = "M003 - 로그인이 필요한 화면입니다."),
	})
	...
	public ResponseEntity<ResultResponse> deletePost(@RequestParam Long postId) {
		...
	}
...
}
```
- 각 API에서 발생할 수 있는 응답들 추가
- 동일 코드인 경우 한 메시지에 줄바꿈으로 추가해야 적용 됨
### 기타
#### 컨벤션 적용
#### ErrorResponse 리팩토링
```diff
public class ErrorResponse {
...
	@Getter
	@NoArgsConstructor(access = AccessLevel.PROTECTED)
	public static class FieldError {
		private static List<FieldError> of(final Set<ConstraintViolation<?>> constraintViolations) {
			List<ConstraintViolation<?>> lists = new ArrayList<>(constraintViolations);
			return lists.stream()
-				.map(error -> new FieldError(
-					error.getPropertyPath().toString(),
-					 "" ,
-					error.getMessageTemplate()))
+				.map(error -> {
+					final String invalidValue = error.getInvalidValue() == null ? "" : error.getInvalidValue().toString();
+					final int index = error.getPropertyPath().toString().indexOf(".");
+					final String propertyPath = error.getPropertyPath().toString().substring(index + 1);
+					return new FieldError(propertyPath, invalidValue, error.getMessage());
				})
				.collect(Collectors.toList());
		}
        }
...
}
```
- `propertyPath`: error.getPropertyPath()는 클래스 이름부터 시작하므로, 첫 번째 `.`을 기준으로 잘라서 이후 문자열(필드명)만 사용
- `invalidValue`: invalidValue가 null인 경우 빈 값을 넣고, 그렇지 않은 경우 입력값을 사용
- 에러 메시지는 템플릿 메시지 적용이 안되어서 기본 메시지로 사용함

<!--
✅ 추가로 전달할 내용이 있다면 적어주세요.
-->
## 💬Comment
-

<!--
✅ 참고한 사이트가 있다면 공유해주세요.
-->
## 📑References
- 

## ✅Check List
- [x] 추가한 기능에 대한 테스트는 모두 완료하셨나요?
- [x] 코드 정렬(Ctrl + Alt + L), 불필요한 코드나 오타는 없는지 확인하셨나요?

seonpilKim committed:
Feat: Swagger 설정 추가

- Authorization 헤더 required = false로 변경
- globalResponseMessage 추가

Resolve: #161

seonpilKim committed:
Refactor: PostController 리팩토링

- @ApiResponse 추가
- 클래스 단위 @RequestMapping("/post") 추가
- 컨벤션 적용
- 불필요 어노테이션 제거(@NotNull)
- @Validated -> @Valid 변경

Resolve: #161

seonpilKim committed:
Refactor: CommentController 리팩토링

- @ApiResponse 추가
- 클래스 단위 @RequestMapping("/comments") 추가
- 컨벤션 적용
- @Validated -> @Valid 변경
- PostController 400 응답 추가

Resolve: #161

seonpilKim committed:
Refactor: ErrorResponse 리팩토링

- ErrorResponse.FieldError.of() 로직 수정
- 컨벤션 적용

seonpilKim committed:
Merge branch 'develop' of https://github.com/Instagram-Clone-Coding/Spring_instagram-clone into Refactor/feed

seonpilKim committed:
Style: feed 도메인 컨벤션 적용

- 클래스 블록 위에 새 줄 추가

vectorch9 commented on src/main/java/cloneproject/Instagram/domain/feed/dto/PostImageTagRequest.java:
```diff
@@ -35,4 +35,5 @@ public class PostImageTagRequest {
 	@ApiModelProperty(value = "게시물 이미지 태그 사용자 아이디", example = "dlwlrma", required = true)
 	@NotBlank(message = "게시물 이미지 태그 사용자 아이디는 필수입니다.")
 	private String username;
+
 }
```
파일 끝에 빈줄이 없는 것 같습니다 !

vectorch9 commented on src/main/java/cloneproject/Instagram/domain/feed/dto/MiniProfilePostDTO.java:
```diff
@@ -22,4 +22,5 @@ public MiniProfilePostDTO(Long postId, String postImageUrl) {
 		this.postId = postId;
 		this.postImageUrl = postImageUrl;
 	}
+
 }
```
여기도 없는 것 같습니다

vectorch9 commented on src/main/java/cloneproject/Instagram/domain/feed/controller/CommentController.java:
```diff
@@ -10,33 +10,54 @@
+			+ "F012 - 댓글 기능이 해제된 게시물에는 댓글을 작성할 수 없습니다.\n"
+			+ "F013 - 최상위 댓글에만 답글을 업로드할 수 있습니다."),
+		@ApiResponse(code = 401, message = "M003 - 로그인이 필요한 화면입니다."),
+	})
+	@PostMapping
+	public ResponseEntity<ResultResponse> createComment(@Valid @RequestBody CommentUploadRequest request) {
```
`@PostMapping("/")`이 좀 더 통일된 느낌일 것 같습니다 !

seonpilKim replied:
제 생각은 좀 다른데, `"/"`은 뒤에 추가적인 path가 있을 때 사용하면 좋을 것 같아요.

vectorch9 commented on src/main/java/cloneproject/Instagram/domain/feed/repository/PostImageRepository.java:
```diff
@@ -23,4 +23,5 @@ public interface PostImageRepository extends JpaRepository<PostImage, Long>, Pos
 		+ "from PostImage pi "
 		+ "where pi.post.id in :postIds")
 	List<PostImageDTO> findAllPostImageDto(@Param(value = "postIds") List<Long> postIds);
+
```
저번에 제안한 메서드 명에 들어가는 DTO에 관한 컨벤션을 적용해야 할 것 같습니다.
- Dto, DTO로 혼용중인 내용을 DTO로 통일
- DTO의 List를 반환하는 경우 메서드명을 DTOs로 적용

추가로, List를 반환 하는 경우 DTO가 아니더라도 모두 s를 붙이는게 좋을거 같은데 어떻게 생각하시나요?

seonpilKim replied:
수정하면서 `findAllPostLikeDTOOfFollowings` 이런 경우에는 가독성이 떨어지는 것 같다는 생각이 들었어요.
DTO 대신 Dto로 사용하는 게 좋을 것 같다는 생각이 드네요
repository에서 리스트를 반환하는 메소드 명은 `findAll~`으로 통일하는 게 좋을 것 같습니다.

seonpilKim committed:
Style: 컨벤션 적용

- DTO -> Dto 네이밍 변경
- 일부 Response 필드명 변경
  - PostDto: postImageDTOs -> postImages
  - PostImageDto: postTagDTOs -> postTags
  - MemberPostDto: postImageDTO -> postImages
  - SearchMemberDto: memberDTO -> member
- 클래스 끝 블록 위에 새 줄 추가
- 값 변경이 되지 않는 변수 final 키워드 추가

seonpilKim commented:
@bluetifulc 
최근 커밋 확인 부탁드립니다.
feed 도메인과 연관 있는 파일까지 함께 컨벤션 적용했습니다.

가독성 문제도 있고, 카멜 케이스에 맞게 DTO 보다는 Dto로 사용하는게 좋을 것 같아서 수정했습니다. 

JPA 표준 메소드 네이밍과 동일하게, 리포지토리에서 데이터를 가져올 때 메소드명은 `find~~`으로 통일하는 게 좋을 것 같습니다. 컬렉션 조회는 `findAll~~` -> `get~~` 메소드명으로 작성하신 코드 있으시면 수정 부탁드립니다!

Response 필드명에 Dto보다는 다른 의미있는 이름을 사용하는 게 좋을 것 같습니다.
feed 도메인과 연관있는 몇몇 파일은 수정해 두었는데, 다른 파일들도 있다면 수정 부탁드립니다.
Response 필드명이 변경되어서 프론트분들께도 전달해야 합니다. -> 제가 수정한 건 내일 카톡으로 전달하겠습니다.

vectorch9 commented:
추가적인 컨벤션은 종강 후 적용하겠습니다
